### ISSUE TITLE
ERROR: failed to solve: process "/bin/sh -c ./build.sh ${COMMON_BUILD_ARGS} --update --build  --allow_running_as_root --use_openvino CPU_FP32" did not complete successfully:

### ISSUE THREAD

**Issue by vkadlec** [triton-inference-server/server#6613]
Hi, I am trying to build Triton Interference Server for CPU-only but error I have error during comparing values in eigen-3.4..zip

I am using branch 23.09...

Does anyone know what to do, please?

Thanks!!

44.23 -- Populating eigen
44.25 -- Configuring done (0.0s)
44.26 -- Generating done (0.0s)
44.26 -- Build files have been written to: /workspace/build/Release/_deps/eigen-subbuild
44.28 [ 11%] Creating directories for 'eigen-populate'
44.29 [ 22%] Performing download step (download, verify and extract) for 'eigen-populate'
44.30 -- Downloading...
44.30    dst='/workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip'
44.30    timeout='none'
44.30    inactivity timeout='none'
44.30 -- Using src='https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip'
44.97 -- verifying file...
44.97        file='/workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip'
44.98 -- SHA1 hash of
44.98     /workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip
44.98   does not match expected value
44.98     expected: 'ee201b07085203ea7bd8eb97cbcb31b07cfa3efb'
44.98       actual: '90414f534834cb5041a4c5ddece24978033d87c7'
44.98 -- Hash mismatch, removing...
44.98 -- Using src='https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip'
45.29 -- verifying file...
45.29        file='/workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip'
45.31 -- SHA1 hash of
45.31     /workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip
45.31   does not match expected value
45.31     expected: 'ee201b07085203ea7bd8eb97cbcb31b07cfa3efb'
45.31       actual: '90414f534834cb5041a4c5ddece24978033d87c7'
45.31 -- Hash mismatch, removing...
45.31 -- Using src='https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip'
45.65 -- verifying file...
45.65        file='/workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip'
45.66 -- SHA1 hash of
45.66     /workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip
45.66   does not match expected value
45.66     expected: 'ee201b07085203ea7bd8eb97cbcb31b07cfa3efb'
45.66       actual: '90414f534834cb5041a4c5ddece24978033d87c7'
45.66 -- Hash mismatch, removing...
45.66 -- Using src='https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip'
46.18 -- verifying file...
46.18        file='/workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip'
46.19 -- SHA1 hash of
46.19     /workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip
46.19   does not match expected value
46.19     expected: 'ee201b07085203ea7bd8eb97cbcb31b07cfa3efb'
46.19       actual: '90414f534834cb5041a4c5ddece24978033d87c7'
46.19 -- Hash mismatch, removing...
46.19 -- Using src='https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip'
46.76 -- verifying file...
46.76        file='/workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip'
46.77 -- SHA1 hash of
46.77     /workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip
46.77   does not match expected value
46.77     expected: 'ee201b07085203ea7bd8eb97cbcb31b07cfa3efb'
46.77       actual: '90414f534834cb5041a4c5ddece24978033d87c7'
46.77 -- Hash mismatch, removing...
46.77 -- Using src='https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip'
47.26 -- verifying file...
47.26        file='/workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip'
47.28 -- SHA1 hash of
47.28     /workspace/build/Release/_deps/eigen-subbuild/eigen-populate-prefix/src/eigen-3.4.zip
47.28   does not match expected value
47.28     expected: 'ee201b07085203ea7bd8eb97cbcb31b07cfa3efb'
47.28       actual: '90414f534834cb5041a4c5ddece24978033d87c7'
47.28 -- Hash mismatch, removing...
47.28 CMake Error at eigen-subbuild/eigen-populate-prefix/src/eigen-populate-stamp/download-eigen-populate.cmake:170 (message):
47.28   Each download failed!
47.28
47.28
47.28
47.28
47.28
47.28 gmake[2]: *** [CMakeFiles/eigen-populate.dir/build.make:100: eigen-populate-prefix/src/eigen-populate-stamp/eigen-populate-download] Error 1
47.28 gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/eigen-populate.dir/all] Error 2
47.28 gmake: *** [Makefile:91: all] Error 2
47.28 CMake Error at /usr/share/cmake-3.27/Modules/FetchContent.cmake:1662 (message):
47.28   Build step for eigen failed: 2
47.28 Call Stack (most recent call first):
47.28   /usr/share/cmake-3.27/Modules/FetchContent.cmake:1802:EVAL:2 (__FetchContent_directPopulate)
47.28   /usr/share/cmake-3.27/Modules/FetchContent.cmake:1802 (cmake_language)
47.28   external/eigen.cmake:21 (FetchContent_Populate)
47.28   external/onnxruntime_external_deps.cmake:428 (include)
47.28   CMakeLists.txt:549 (include)
47.28
47.28
47.29 -- Configuring incomplete, errors occurred!
47.30 Traceback (most recent call last):
  54 |
  55 | >>> RUN ./build.sh ${COMMON_BUILD_ARGS} --update --build  --allow_running_as_root --use_openvino CPU_FP32
  56 |
  57 |     #
--------------------
ERROR: failed to solve: process "/bin/sh -c ./build.sh ${COMMON_BUILD_ARGS} --update --build  --allow_running_as_root --use_openvino CPU_FP32" did not complete successfully: exit code: 1
make[2]: *** [CMakeFiles/ort_target.dir/build.make:74: onnxruntime/lib/libonnxruntime.so] Error 1
make[2]: Leaving directory '/tmp/tritonbuild/onnxruntime/build'
make[1]: *** [CMakeFiles/Makefile2:275: CMakeFiles/ort_target.dir/all] Error 2
make[1]: Leaving directory '/tmp/tritonbuild/onnxruntime/build'
make: *** [Makefile:136: all] Error 2

**Comment by kthui** [comment]
Hi @vkadlec, the hash for eigen-3.4.zip file does not match the one expected. You might want to make sure the file downloaded is the correct one.

Alternatively, you can also use our pre-build container, to skip over any build issues: https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tritonserver/tags

**Comment by vkadlec** [comment]
> Alternatively, you can also use our pre-build container, to skip over any build issues:

Thank you very much for the answer! 

Maybe I should also ask which of the available containers is optimized to run on CPU? That means without unnecessary containers. Sorry for the question...

**Comment by kthui** [comment]
Yes, `nvcr.io/nvidia/tritonserver:23.10-py3`, for example, is the right container.

**Comment by Jackiexiao** [comment]
I have the same problem, the hash for eigen-3.4.zip file does not match the one expected. but why.... should we change hash in build.py?

https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip
```
	python3 build.py  \
		--enable-logging --enable-stats --enable-tracing --enable-metrics \
		--endpoint=http --endpoint=grpc \
		--backend=ensemble \
		--backend=python \
		--backend=onnxruntime
```

**Comment by Jackiexiao** [comment]
ok... onnx had fixed this issue https://github.com/microsoft/onnxruntime/issues/18286

**Comment by Jackiexiao** [comment]
yes, it is the latest code is on the main branch

**Comment by Jackiexiao** [comment]
or you can change 
![image](https://github.com/triton-inference-server/server/assets/18050469/c6f8b985-88c3-403f-82c5-136115ef5c06)

**Comment by dyastremsky** [comment]
Closing issue due to inactivity. Please reopen if you would like to follow up with this issue.
